src/eslint/rules/better-test-assertions.ts

```diff
@@ -0,0 +1,226 @@
+import { AST_NODE_TYPES, ESLintUtils } from '@typescript-eslint/utils';
+
+// https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/developers/Custom_Rules.mdx?plain=1#L109
+
+const createRule = ESLintUtils.RuleCreator<{
+  recommended: boolean;
+}>((name) => {
+  return `https://github.com/CrowdStrike/glide-core/blob/main/src/eslint/rules/${name}.ts`;
+});
+
+export const betterTestAssertions = createRule({
+  name: 'better-test-assertions',
+  meta: {
+    docs: {
+      description:
+        'Ensures assertions are written in a natural, declarative way.',
+      recommended: true,
+    },
+    type: 'suggestion',
+    messages: {
+      preferToBeEmptyString:
+        '`to.be.empty.string` is more declarative. Use it instead.',
+      preferToBeNull: '`to.be.null` is more declarative. Use it instead.',
+      preferToBeUndefined:
+        '`to.be.undefined` is more declarative. Use it instead.',
+      preferToBeTrue: '`to.be.true` is more declarative. Use it instead.',
+      preferToBeFalse: '`to.be.false` is more declarative. Use it instead.',
+      preferToEqual: '`to.equal()` reads better. Use it instead.',
+      preferToDeepEqual: '`to.deep.equal()` reads better. Use it instead.',
+      preferToNotEqual: '`to.not.equal()` reads better. Use it instead.',
+      preferToNotDeepEqual:
+        '`to.not.deep.equal()` reads better. Use it instead.',
+      preferToNotBe: '`to.not.be` reads better. Use it instead.',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    let isProbablyATest = false;
+
+    return {
+      CallExpression(node) {
+        if (
+          node.callee.type === AST_NODE_TYPES.Identifier &&
+          node.callee.name === 'it'
+        ) {
+          isProbablyATest = true;
+        }
+      },
+      MemberExpression(node) {
+        // Covering every Chai property and combination thereof isn't worth our time or
+        // the added code weight. So we instead cover only the most common ones. Similar
+        // for fixing violations automatically. Doing isn't worth the effort or complexity
+        // given how easy these violations are to fix manually.
```

Comment:
Yay? Nay?

Reply:
Could be nice to auto-fix, but the lint message and fixes are pretty straight forward. And this is an improvement over what we currently have, so I think this is good as is, like you said 👍